### DISCUSSION TITLE
Is there a way to not fix a specific line?

### DISCUSSION THREAD

**Discussion by Kludex** [astral-sh/ruff#5213] [Q&A] [answered]
That's it 👆

**Comment by charliermarsh** [comment]
Haha. Not currently. How would this differ from `# noqa: ${code}`? It would continue to raise the violation, but avoid fixing it? What's the use-case?

**Reply by Kludex** [reply]
Typer uses the typing information of the signature at runtime.

See the problem:
```py
from __future__ import annotations

from typer import Typer

app = Typer()

@app.command()
def main(
    param: Union[str, None]  # I don't want to fix this.
):
    return potato(param)
    
def potato(a: str | None) -> str | None:  # I actually want to write like this.
    return a
```

**Reply by charliermarsh** [reply, marked as answer]
I believe `noqa` is what you're looking for but I may be misunderstanding:

```python
from __future__ import annotations

from typer import Typer

app = Typer()

@app.command()
def main(
    param: Union[str, None]  # noqa: UP007
):
    return potato(param)
    
def potato(a: str | None) -> str | None:  # 
    return a
```

**Reply by Kludex** [reply]
No, you are right hahaha

I'm not sure why it formatted some hours ago... 🤔

We'll never know 😎 👍

**Reply by Kludex** [reply]
Thanks.

**Reply by charliermarsh** [reply]
(Are you "allowed" to use `from __future__ import annotations` with Typer?)

**Reply by Kludex** [reply]
(you can, but you can't use a "future" annotation in the signature, since typer will crash...)

Now I got the problem of the discussion again... I mean, I don't have a good use case to defend this, since it's my setup that is the problem.

... But `# fmt: off` and `# fmt: on` would be cool. 👀 

> **Note**
> The problem is that my project structure is weird, and my IDE doesn't get the black rules from the right `pyproject.toml`, so even if the configurations are matching, my IDE runs first black, and then ruff (idk how to change the order in the IDE).
> Anyway... Since black formats in a way that the `# noqa` goes to another line, then I lose the `Union`.
